client/components/utils/add-comment-plugin.js

```diff
@@ -0,0 +1,139 @@
+import {Plugin} from "prosemirror-state";
+
+class AddComment {
+  constructor(view, vueInstance) {
+    view.dom.parentNode.appendChild(vueInstance.$refs.addCommentButton.$el);
+    this.update(view, null);
+    this.vueInstance = vueInstance;
+  }
+
+  /**
+   * Function that is being called with every action from the editor, here we
+   * check if a selection was made and show the "Add Comment" button.
+   * @param {F} view - EditorView from ProseMirror
+   * @param {*} lastState - Previous State
+   */
+  update(view, lastState) {
+    const {state} = view;
+    const {selection} = state;
+    if (
+      (lastState &&
+      lastState.doc.eq(state.doc) &&
+      lastState.selection.eq(state.selection)) ||
+      !this.vueInstance) {
+      return;
+    }
+
+    const marks = [];
+    state.doc.nodesBetween(selection.from, selection.to, (node, start, parent, index) => {
+      marks.push({marks: node.marks, start, size: node.nodeSize});
+    });
+    marks.shift(); // the first element always comes empty, so we remove it
+    let onlyHighlightMarkInRange = true;
+    marks.forEach((marksObj) => {
+      if (!marksObj.marks.length) {
+        onlyHighlightMarkInRange = false;
+      }
+      if (marksObj.marks.filter((m) => {
+        return m.type.name !== 'highlight';
+      }).length) {
+        onlyHighlightMarkInRange = false;
+      }
+    });
+    this.vueInstance.selectedExistingComments = marks.filter((marksObj) => {
+      return marksObj.marks.length;
+    }).map((marksObj) => {
+      return Object.assign({}, marksObj, {
+        marks: marksObj.marks.filter((m) => {
+          return m.type.name === "highlight";
+        }),
+      });
+    });
+    const button = this.vueInstance.$refs.addCommentButton;
+    // Hide the comment button if the selection is empty or the selection
+    // only contains highlight marks.
+    if (state.selection.empty || onlyHighlightMarkInRange) {
+      button.$el.style.opacity = 0;
+      button.$el.style.visibility = 'hidden';
+      return;
+    }
+
+    button.$el.style.opacity = 0.75;
+    button.$el.style.visibility = 'visible';
+    const {from} = state.selection;
+    // These are in screen coordinates
+    const start = view.coordsAtPos(from);
+    // The box in which the comment button is positioned, to use as base
+    const box = button.$el.offsetParent.getBoundingClientRect();
+    button.$el.style.bottom = `${(box.bottom - start.bottom)}px`;
+  }
+}
+
+export default function addCommentPlugin(vueInstance) {
+  return new Plugin({
+    view(editorView) {
+      return new AddComment(editorView, vueInstance);
+    },
+  });
+}
+
+export function addHighlight(keys, schema, state, from, to, dispatch) {
+  const {doc, tr} = state;
+  let attrs = null;
+  if (!doc.rangeHasMark(from, to, schema.marks.highlight)) {
+    attrs = {"data-highlight-keys": keys};
+    if (!attrs["data-highlight-keys"]) {
```

Comment:
I do not get this check. What are you checking here?

Reply:
I think that is a leftover from copying how link marks work. I think I can remove it and see if it's still working fine, since it doesn't really make sense. Where should I try and push? a new PR?

Reply:
Yes, please try. It can be a short PR.